scispacy/abbreviation.py

```diff
@@ -96,26 +96,36 @@ def filter_matches(
             # Long form is inside the parens.
             # Take one word before.
             short_form_candidate = doc[start - 2 : start - 1]
-            if short_form_filter(short_form_candidate):
-                candidates.append((doc[start:end], short_form_candidate))
+            long_form_candidate = doc[start:end]
         else:
             # Normal case.
             # Short form is inside the parens.
+            short_form_candidate = doc[start:end]
+
             # Sum character lengths of contents of parens.
-            abbreviation_length = sum([len(x) for x in doc[start:end]])
+            abbreviation_length = sum([len(x) for x in short_form_candidate])
             max_words = min(abbreviation_length + 5, abbreviation_length * 2)
             # Look up to max_words backwards
             long_form_candidate = doc[max(start - max_words - 1, 0) : start - 1]
-            candidates.append((long_form_candidate, doc[start:end]))
+
+        # add candidate to candidates if candidates pass filters
+        if short_form_filter(short_form_candidate):
+            candidates.append((long_form_candidate, short_form_candidate))
+
     return candidates
 
 
 def short_form_filter(span: Span) -> bool:
     # All words are between length 2 and 10
-    if not all([2 < len(x) < 10 for x in span]):
+    if not all([2 <= len(x) < 10 for x in span]):
```

Comment:
I believe strictly greater than 2 here is motivated by one of your xfail cases, namely that references, citations etc are often single or 2 char identifiers, like `(1), [2], (iv)` etc. But fine if it catches some case you need!

Reply:
yeah, the thing is that previously the short form filter was only being applied in the <short> (<long>) case. When applied in both cases, it even fails one of the tests we have, which contains a two char abbreviation. In general, I think there can definitely be short forms that are two chars. I think whatever extra ones show up like (iv) will be easier to filter in an end use case than it would be to capture missing two char abbreviations so i'm going to go ahead and merge this.